include/exadg/grid/calculate_characteristic_element_length.h

```diff
@@ -55,16 +59,55 @@ calculate_minimum_vertex_distance(dealii::Triangulation<dim> const & triangulati
   return dealii::Utilities::MPI::min(min_cell_diameter, mpi_comm);
 }
 
+/**
+ * This function calculates a characteristic element length given the volume of the element, the
+ * number of space dimensions, and the element type.
+ */
+template<typename Number>
+inline Number
+calculate_characteristic_element_length(Number const        element_volume,
+                                        unsigned int const  dim,
+                                        ElementType const & element_type)
+{
+  if(element_type == ElementType::Hypercube)
+  {
+    // calculate the length h of a dim-dimensional cube that has the volume "element_volume".
+    return std::exp(std::log(element_volume) / (double)dim);
+  }
+  else if(element_type == ElementType::Simplex)
+  {
+    // calculate the length h of an equilateral triangle/tetrahedron that has the volume
+    // "element_volume" (V = h^dim / factor).
+    double factor = 1.0;
+
+    if(dim == 2)
+      factor = 4.0 / std::sqrt(3.0);
+    else if(dim == 3)
+      factor = 6.0 * std::sqrt(2.0);
+    else
+      AssertThrow(false, dealii::ExcMessage("Only dim = 2, 3 implemented."));
+
+    return std::exp(std::log(element_volume * factor) / (double)dim);
+  }
```

Comment:
As far as I know, we have for
`dim==2`: `A=a^2 * sqrt(3)/4`, so, all good,
but for
`dim==3`: `V=a^3 * sqrt(2)/12`, see, e.g., [link](https://keisan.casio.com/exec/system/1223345456) or [link](https://www.mathematische-basteleien.de/tetrahedron.htm)
which would give a different factor for the three-dimensional case?

Reply:
Isn't `(sqrt(2)/12)^{-1}` the same as `6 * sqrt(2)`?

Reply:
seems I need a coffee ...